dev-packages/node-integration-tests/utils/runner.ts

```diff
@@ -187,33 +187,110 @@ export function createEsmAndCjsTests(
     throw new Error(`Instrument file not found: ${mjsInstrumentPath}`);
   }
 
-  const cjsScenarioPath = join(cwd, `tmp_${scenarioPath.replace('.mjs', '.cjs')}`);
-  const cjsInstrumentPath = join(cwd, `tmp_${instrumentPath.replace('.mjs', '.cjs')}`);
+  // If additionalDependencies are provided, we create a dedicated tmp directory that includes
+  // copied ESM & CJS scenario/instrument files and a nested package.json with those dependencies installed.
+  const useTmpDir = Boolean(options?.additionalDependencies && Object.keys(options.additionalDependencies).length > 0);
```

Comment:
Hmm, what about simply always creating a tmp dir? Would unify this a bit, I suppose...?